### PR TITLE
Fixes issue where sample selection is not maintained between table pages.

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
There appears to be a change in the pay redux toolkit returns stat within a slice `addCase` callback, it now appears to return a proxy to the state, which was causing issues comparing objects.   This was causing sample selection across pages to become broken on the project samples table.

I fixed this issue by moving the logic up into an `createAsyncThunk` function where I can get the state directly using the `getState` method.  Instead of just strinifying the object, I updated it to use the lodash isEqual function.

## Related issue
N/A

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~ 
  - Not required as this is a development issue.

* [X] Tests added (or description of how to test) for any new features.

~* [ ] User documentation updated for UI or technical changes.~
  - Not required an no UI or functional changes

### CODE REVIEW

File: src/main/webapp/resources/js/pages/projects/redux/samplesSlice.js [hunk 1]
```diff
@@ -3,21 +3,49 @@ import {
...
+ * the re-render of the samples table.
+ * @type {AsyncThunk<unknown, void, {}>}
+ */
+const updateTable = createAsyncThunk(
+  "samples/table/update",
+  async (values, { getState }) => {
+    const {
+      samples: { options, selected, selectedCount },
+    } = getState();
+    console.log("SEARCH", { state: options.search, values: values.search });
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/redux/samplesSlice.js:34]: Remove the logging statements here and below

File: src/main/webapp/resources/js/pages/projects/samples/components/SamplesTable.jsx [hunk 2]
```diff
@@ -43,33 +43,31 @@ export function SamplesTable() {
...
 
   /**
    * Handle row selection change event
    * @param event
    * @param sample
    */
   const onRowSelectionChange = (event, sample) => {
-    const selected = event.target.checked;
-    if (selected) {
+    console.log(selected);
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/samples/components/SamplesTable.jsx:69]: Remove logging statement

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good. Just a couple of things below

File: src/main/webapp/resources/js/pages/projects/redux/samplesSlice.js (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/redux/samplesSlice.js]: Removed in d73f8bb385198793f63345d399f52a4226e6be79

File: src/main/webapp/resources/js/pages/projects/samples/components/SamplesTable.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/samples/components/SamplesTable.jsx:69]: Removed in d73f8bb385198793f63345d399f52a4226e6be79

Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/projects/samples/components/SamplesTable.jsx:69]: The logging statement is still in this file

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]

**Comment by joshsadam** [comment]
Removed that loggin statement in 643eb5a2eb9d3e8d8f0bd1603af5a8b7ef687f69, thanks

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**